From 24eb4503866b8f7ffe9ee534378365521fc4ea2e Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Sun, 06 Feb 2022 15:40:28 +0200
Subject: [PATCH] [SW-56024] fwupd: fix translate adjust mask

Mask wasn't masking enough of full address bit 26 allowed access to
128MB memory space, flash has only 64MB memory space.

Signed-off-by: Noam Lessner <nlessner@habana.ai>
Change-Id: I982b11e1078ab6f052619198e9bf5d372438fd70
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 4ebe54a..8dac81d 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -490,7 +490,7 @@
 	}
 
 	if (addr >= SDRAM_FULL_BASE && addr < (SDRAM_FULL_BASE + (u64)ARM_SDRAM_SIZE)) {
-		addr = (addr & ARC_ADDR_MASK) | BIT(28);
+		addr = (addr & ARC_SDRAM_ADDR_MASK) | BIT(28);
 		goto ret_addr;
 	}
 
diff --git a/arch/gaudi2/arc/common.h b/arch/gaudi2/arc/common.h
index efadc6a..835271f 100644
--- a/arch/gaudi2/arc/common.h
+++ b/arch/gaudi2/arc/common.h
@@ -104,6 +104,7 @@
 #define SPI_DMA_BAUDR_SRAM_DMA	0x4
 
 #define ARC_ADDR_MASK		0x7FFFFFFull
+#define ARC_SDRAM_ADDR_MASK	0x3FFFFFFull
 
 #define sleep()	__asm__ __volatile__("sleep\n\t" : : )
 
